addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/InsecureComponentScannerUnitTest.java

```diff
@@ -0,0 +1,408 @@
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Server: PHP/2.6.0");
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+        // Then
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void shouldRaiseVulnerabilityAlertWhenApacheWebServerFromRedHatUsed()
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Server: Apache/1.5.0 (Red Hat)");
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+        // Then
+        assertVulnerabilityAlertRaised(Alert.RISK_MEDIUM, "Apache/1.5.0 (Red Hat)");
+        assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_LOW));
+    }
+
+    @Test
+    public void shouldNotRaiseVulnerabilityAlertWhenApacheWebServerFromUbuntuUsed()
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Server: Apache/1.5.0 (Ubuntu)");
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+        // Then
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void shouldRaiseVulnerabilityAlertWhenInsecurePerlVersionAsApacheWebServerModuleUsed()
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Server: Apache/1.5.0 Perl/v4.0.0");
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+        // Then
+        assertVulnerabilityAlertRaised(Alert.RISK_MEDIUM, "Apache/1.5.0 Perl/v4.0.0");
+    }
+
+    @Test
+    public void shouldNotRaiseVulnerabilityAlertWhenSecurePerlVersionAsApacheWebServerModuleUsed()
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Server: Apache/1.5.0 Perl/v4.0.1");
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+        // Then
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void shouldRaiseVulnerabilityAlertWhenInsecureContentGeneratorUsed()
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Server: Apache/1.5.0 Perl/v4.0.1");
+        String metaTag = "<meta name=\"generator\" content=\"Frontweaver 1.2.0\"/>";
+        msg.setResponseBody("<html><head>" + metaTag + "</head></html>");
+
```

Comment:
The blank lines preceding comments here and below could be removed.

Reply:
The blank lines are removed.